scipy/optimize/_cobyla_py.py

```diff
@@ -218,6 +218,29 @@ def _minimize_cobyla(fun, x0, args=(), constraints=(),
     if isinstance(constraints, dict):
         constraints = (constraints, )
 
+    if bounds:
```

Comment:
I wonder if we should have a generic `scipy.optimize` function for doing this bounds checking?

Reply:
Probably. I was sad that it wasn't there already : ( Coming in next commmit.